src/test/rpc/AccountTx_test.cpp

```diff
@@ -753,6 +755,93 @@ class AccountTx_test : public beast::unit_test::suite
+        Account const alice{"alice"};
+        Account const bob{"bob"};
+        Account const carol{"carol"};
+        std::optional<uint256> ledgerHash1;
+        std::optional<uint256> ledgerHash2;
+
+        // check the latest mpt-related txn is in alice's account history
+        auto const checkAliceAcctTx =
+            [&](Env& env, size_t size, Json::StaticString txType) {
+                Json::Value params;
+                params[jss::account] = alice.human();
+                params[jss::limit] = 100;
+                auto const jv = env.rpc(
+                    "json", "account_tx", to_string(params))[jss::result];
+
+                BEAST_EXPECT(jv[jss::transactions].size() == size);
+                auto const& tx0(jv[jss::transactions][0u][jss::tx]);
+                BEAST_EXPECT(tx0[jss::TransactionType] == txType);
+            };
+
+        {
+            Env env(*this, features | featureMPTHistory);
+
+            MPTTester mptAlice(env, alice, {.holders = {bob, carol}});
+
+            // alice creates issuance
+            mptAlice.create(
+                {.ownerCount = 1,
+                 .holderCount = 0,
+                 .flags =
+                     tfMPTCanClawback | tfMPTRequireAuth | tfMPTCanTransfer});
+
+            checkAliceAcctTx(env, 3, jss::MPTokenIssuanceCreate);
+
+            // bob creates a MPToken;
+            // with featureMPTHistory, this tx should show up in alice's account
+            // history
+            mptAlice.authorize({.account = bob});
+            checkAliceAcctTx(env, 4, jss::MPTokenAuthorize);
+
+            ledgerHash1 = env.closed()->info().hash;
+        }
+
+        {
+            Env env(*this, features - featureMPTHistory);
+
+            MPTTester mptAlice(env, alice, {.holders = {bob, carol}});
+
+            // alice creates issuance
+            mptAlice.create(
+                {.ownerCount = 1,
+                 .holderCount = 0,
+                 .flags =
+                     tfMPTCanClawback | tfMPTRequireAuth | tfMPTCanTransfer});
+
+            checkAliceAcctTx(env, 3, jss::MPTokenIssuanceCreate);
+
+            // bob creates a MPToken;
+            // without featureMPTHistory, this tx should NOT show up in alice's
+            // account history
+            mptAlice.authorize({.account = bob});
+            checkAliceAcctTx(env, 3, jss::MPTokenIssuanceCreate);
+
+            ledgerHash2 = env.closed()->info().hash;
+        }
+
+        std::cout << "ledgerHash1 " << to_string(ledgerHash1.value())
+                  << std::endl;
+        std::cout << "ledgerHash2 " << to_string(ledgerHash2.value())
+                  << std::endl;
+
+        auto const expectedLedgerHash =
+            "1D6B581B977FA443FDD2419CF8E98199170F5C6887661F5908E4D76A5270EC7C";
+
+        BEAST_EXPECT(to_string(ledgerHash1.value()) == expectedLedgerHash);
+        BEAST_EXPECT(to_string(ledgerHash2.value()) == expectedLedgerHash);
+        BEAST_EXPECT(ledgerHash1 == ledgerHash2);
```

Comment:
the same ledgerhash from two identical tests where one has the amendment enabled and one does not.